tests/integration_tests/test_upgrade.py

```diff
@@ -40,6 +41,19 @@ def _output_to_compare(instance, file_path, netcfg_path):
             f.write(instance.execute(command) + '\n')
 
 
+def _restart(instance):
+    # work around pad.lv/1890528
```

Comment:
Could you explain in a little more detail how this is doing that? My reading of that bug is that it only triggers if `cloud-init status --wait` exits non-zero: if that's exiting non-zero (and therefore raising the `OSError`, I assume?) then `result.ok` in the `for` loop below is never going to be `True`, so we'll never `return` early. What am I missing?

(Even if I'm not missing something and this is a bug, this doesn't need to block landing; we should fix it soon hereafter though.)

Reply:
Hmmm, maybe this is a different bug then. On ec2, cloud-init returns early with non-zero status (I'm inferring this from pycloudlib behavior). Connecting again, it'll complete and return 0. Nothing in the logs or result.json or status.json indicate any sort of failure or problem anywhere.

Reply:
I'll file a separate bug with logs and update the reference here

Reply:
Thanks!